### DISCUSSION TITLE
Does centered_float widgets keep listening to scope events after close?

### DISCUSSION THREAD

**Discussion by serranomorante** [mfussenegger/nvim-dap#1171] [Q&A] [answered]
Using this mapping was slowing down my neovim instance by a lot.

```lua
local widgets = require("dap.ui.widgets")
local scopes_sidebar = widgets.sidebar(widgets.scopes, { number = true, wrap = false })
vim.keymap.set("n", "<leader>ds", scopes_sidebar.toggle, { desc = 'DAP: Toggle "scopes" in sidebar' })
```

On the other hand, this one seems to perform better.

```lua
...
      function()
        local ui = require("dap.ui.widgets")
        ui.centered_float(ui.scopes, { number = true, wrap = false })
      end,
...
```

I just wanted to ask what is the lifetime of a `centered_float` view, does it keep listening to dap events after closing the floating window?

Also, any ideas on how to better debug the performance of the first keymap?

I believe I might be able to replicate the issue.

Thank you!

**Comment by mfussenegger** [comment, marked as answer]
The `centered_float` doesn't add any listener at all - it also won't refresh while it's vissible.

The point about the sidebar always listening is a good one. I should tune that to only listen while the view is open. 

To clarify: the slowdown only happens when a breakpoint is hit, or in general? The listeners are only triggered if the scopes get refreshed (which then redraws all the variables)

**Reply by serranomorante** [reply]
> To clarify: the slowdown only happens when a breakpoint is hit, or in general? The listeners are only triggered if the scopes get refreshed (which then redraws all the variables)

It starts to happen gradually from the first breakpoint being hit and gets worse and worse overtime in the same debug session while keeping the scopes sidebar opened. This is happening to me while debugging a react (javascript) web project.

Thank you for answering, I will keep just using `centered_float` for now and maybe try to replicate the first behavior and opening an issue 👍

**Reply by serranomorante** [reply]
I did replicate the lag issue here: https://github.com/mfussenegger/nvim-dap/issues/1176
